src/main/java/com/internet/shop/factory/Factory.java

```diff
@@ -0,0 +1,15 @@
+package com.internet.shop.factory;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.dao.impl.ProductDaoImpl;
+
+public class Factory {
+    private static ProductDao productDao;
+
+    public static ProductDao getProductDao() {
+        if (productDao == null) {
+            productDao = new ProductDaoImpl();
+        }
+        return productDao;
+    }
+}
```

Comment:
do we need this class? this work is done by the Injector

Reply:
Done